### ISSUE TITLE
SMTP relay password should be validated to ensure there are no # char in it

### ISSUE THREAD

**Issue by jedie** [YunoHost/issues#1800]
It seems that the **smtp.relay.password** must be ASCII only. 
See: https://forum.yunohost.org/t/setup-smtp-relay/15776

**Comment by zamentur** [comment]
In fact it can't contains # as # is the comment characters.

See this https://linuxaria.com/pills/how-to-configure-postfix-as-smtp-relay-for-gmail-on-archlinux#comment-5072

If we inspect the sasl_passwd format we can imagine it's not easy to be sure that ":" is supported...
https://github.com/YunoHost/yunohost/blob/02a30125b578f22238ef2c4bae45f3eec6a9d94d/data/hooks/conf_regen/19-postfix#L45

**Comment by jedie** [comment]
Let me put it this way: Regardless of which characters are really not allowed: It should basically not be possible to use unallowed characters. There should always be an error message that is as self-explanatory as possible.

The hint in the documentation is only what I can contribute so quickly.

**Comment by zamentur** [comment]
> Regardless of which characters are really not allowed: It should basically not be possible to use unallowed characters. There should always be an error message that is as self-explanatory as possible.

I agree with you however when this feature was created we was not aware of this limitation opf postfix sasl_passwd... Thats' why there is no check about it.
